providers/hubspot/errors.go

```diff
@@ -52,11 +52,7 @@ type ErrLinks struct {
 
 func (c *Connector) interpretError(res *http.Response, body []byte) error {
 	mediaType, _, err := mime.ParseMediaType(res.Header.Get("Content-Type"))
-	if err != nil {
-		return fmt.Errorf("mime.ParseMediaType failed: %w", err)
-	}
-
-	if mediaType == "application/json" {
+	if err == nil && mediaType == "application/json" {
```

Comment:
I can create a follow up PR but hubspot lacks mock tests to make such changes and I don't want to break prod.

I reviewed the code and the error handling can be replaced like so:
```
conn.Client.HTTPClient.ErrorHandler = interpreter.ErrorHandler{
	JSON: interpreter.NewFaultyResponder(errorFormats, statusCodeMapping),
}.Handle


var errorFormats = interpreter.NewFormatSwitch(
	[]interpreter.FormatTemplate{
 		{
			MustKeys: nil,
			Template: func() interpreter.ErrorDescriptor { return &HubspotError{} },
		},
	}...,
)

var statusCodeMapping = map[int]error{
	http.StatusBadGateway:      common.ErrLimitExceeded,
        ....
}
```